### PR TITLE
Adding Snowflake's public documentation

### PR DESCRIPTION
**Pull request by scottteal**
Updating vendor description links with latest public documentation from Snowflake. Adding Snowflake's primary docs link for Iceberg to the sidebar.

### CODE REVIEW

File: docs/config.toml [hunk 1]
```diff
@@ -78,11 +78,12 @@ home = [ "HTML", "RSS", "SearchIndex" ]
...
-    { name = "Doris", identifier = "_doris", weight = 1001, url = "https://doris.apache.org/docs/dev/lakehouse/multi-catalog/iceberg" },
-    { name = "PyIceberg", identifier = "_python", weight = 1500, url = "https://py.iceberg.apache.org/"},
-    { name = "Integrations", weight = 1100 },
-    { name = "API", weight = 1200},
-    { name = "Migration", weight = 1300},
-    { name = "Javadoc", pre = "relative", url = "../../javadoc/latest", weight = 1400} # `url` is populated by the github deploy workflow and is equal to "../../javadoc/<branch name>"
+    { name = "Snowflake", identifier = "_snowflake", weight = 1000, url = "https://docs.snowflake.com/en/user-guide/tables-iceberg" },
+    { name = "Impala", identifier = "_impala", weight = 1100, url = "https://impala.apache.org/docs/build/html/topics/impala_iceberg.html" },
+    { name = "Doris", identifier = "_doris", weight = 1200, url = "https://doris.apache.org/docs/dev/lakehouse/multi-catalog/iceberg" },
+    { name = "PyIceberg", identifier = "_python", weight = 1700, url = "https://py.iceberg.apache.org/"},
```
Comment (Fokko) [inline docs/config.toml:84]: nit: While at it, should we move PyIceberg to the bottom, since it has the highest weight?

**Review by Fokko — APPROVED** [review]
Looks good to me! Thanks @scottteal

File: docs/config.toml (hunk 1 above)
Comment (scottteal) [reply docs/config.toml:84]: Yeah, I couldn't tell what the ordering logic was, also why weights were what they were. Thanks for the cleanup.

**Comment by Fokko** [comment]
@scottteal sorry for the late reply here, could you fix the conflicts once more?

**Comment by scottteal** [comment]
@Fokko I think I've now resolved the conflicts. I tried to preserve the ordering to match the current website as much as possible, although it's still unclear to me what the ordering logic is. They aren't sorted alphabetically, or grouped in a logical manner that I can tell.

**Comment by scottteal** [comment]
@Fokko do you know when this will be merged and deployed?

**Comment by Fokko** [comment]
Sorry for the long wait @scottteal

**Pull request merged**